server/serverAPI/controller/AuthenticationController.ts

```diff
@@ -74,21 +76,51 @@ export default class AuthenticationController extends BaseUserController {
                 username: user.username
             };
 
-            let token = this.tokenCreator.sign(identification, this.timeoutTimeInSeconds);
+
+            let accessToken = this.tokenCreator.sign(identification, this.accessTokenTimeoutInSeconds);
+            let refreshToken = this.tokenCreator.sign(identification, this.refreshTokenTimeoutInSeconds);
+
+            JWTStorage.getInstance().addJWT(
+                userCredentials.username,
+                new Token(accessToken, refreshToken)
+            );
 
             if (req.query.includeInfo === 'true') {
                 return this.sendSuccess(200, res, {
-                    accessToken: token,
+                    accessToken: accessToken,
+                    refreshToken: refreshToken,
                     userInfo: this.convertToUserResponse(user)
                 });
             } else {
-                return this.sendSuccess(200, res, { accessToken: token });
+                return this.sendSuccess(200, res, {
+                    accessToken: accessToken,
+                    refreshToken: refreshToken
+                });
             }
         } catch (e) {
             return e;
         }
     }
 
+    refreshToken = async (req: Request, res: Response) => {
+        try {
+            let identification = this.tokenCreator.verify(req.body.refreshToken);
```

Comment:
After verifying that refresh token is valid, I think the refresh token should be refreshed also. Otherwise there's not really a point in having the refresh token, since it's basically just an access token with a longer timeout.

Reply:
Like the whole idea is that it has a longer timeout but still can be invalidated.

Reply:
Actually.... Let's also refresh it, and I will also create a logout endpoint that will destroy all tokens issued. This will calm my worries regarding leaking refreshToken.